akka-actor-tests/src/test/scala/akka/io/dns/AsyncDnsResolverIntegrationSpec.scala

```diff
@@ -97,6 +98,29 @@ class AsyncDnsResolverIntegrationSpec extends AkkaSpec(s"""
         InetAddress.getByName("fd4d:36b2:3eca:a2d8:0:0:0:5"))
     }
 
+    "resolve single A record from IPv4-Mapped address" in {
+      val name = "mapped-single.foo.test"
+      val answer = resolve(name)
+      withClue(answer) {
+        answer.name shouldEqual name
+        answer.records.size shouldEqual 1
+        answer.records.head.name shouldEqual name
+        answer.records.head.asInstanceOf[ARecord].ip shouldEqual InetAddress.getByName("192.168.1.25")
```

Comment:
Hmm, it seems a bit strange to return an `ARecord` when the actual record was an AAAA record.

Reply:
It was 28 code in the DNS response but under the cover java dns checks if it's IPv4 mapped and returns IPv4 if so.
So, I followed the logic and return the same - ARecord, as ARecord can not be cats to AAAARecord.

Reply:
Also, for me it's more clear to have IPv4 for such IPs like ::ffff:192.168.1.25